src/Client/MultiplexedConnections.cpp

```diff
@@ -187,12 +190,17 @@ void MultiplexedConnections::disconnect()
     }
 }
 
-void MultiplexedConnections::sendCancel()
+void MultiplexedConnections::sendCancel(bool ignore_error)
 {
     std::lock_guard lock(cancel_mutex);
 
     if (!sent_query || cancelled)
-        throw Exception("Cannot cancel. Either no query sent or already cancelled.", ErrorCodes::LOGICAL_ERROR);
+    {
+        if (ignore_error)
+            return;
+        else
+            throw Exception("Cannot cancel. Either no query sent or already cancelled.", ErrorCodes::LOGICAL_ERROR);
+    }
```

Comment:
Suspicious change.
At least, we should return some status if we did not send cancel. Is it possible that we ignore error after cancel, but then drain connection (and wait for receive_timeout)?
Maybe, we can add `bool trySendCancel()`